portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,29 +59,89 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    int maxComments = getMaxCommentsToReturn(request);
-
+    JsonObject json = new JsonObject();
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
+      String name = getUserName(userService.getCurrentUser().getUserId());
+
+      // If the user has logged in for the first time, set their name as their Google account nickname.
+      if (name == null) {
+        name = userService.getCurrentUser().getNickname();
+        NameServlet.upsertUserInfo(name);
+      }
+
+      String urlToRedirectToAfterUserLogsOut = "/";
```

Comment:
nit: since the redirect url is actually the same for log out and log in, maybe just create a single variable `redirectUrl = "/";` or `urlToRedirectTo = "/";`